### DISCUSSION TITLE
Cannot dump database

### DISCUSSION THREAD

**Discussion by DaliborPan** [planetscale/discussion#538] [Bug Reports] [answered]
Hi, I'm trying to dump my database using CLI `pscale database dump [name] [branch]`. I was able to do that on 5th September. Now I'm getting errors

```
⠸ Dumping tables ...2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
⠼ Dumping tables ...2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
⠴ Dumping tables ...2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
⠦ Dumping tables ...2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:15+02:00	error	error dumping table	{"error": "EOF"}
⠧ Dumping tables ...2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "write tcp 127.0.0.1:53455->127.0.0.1:53418: use of closed network connection"}
2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "EOF"}
2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "write tcp 127.0.0.1:53457->127.0.0.1:53418: use of closed network connection"}
2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "write tcp 127.0.0.1:53459->127.0.0.1:53418: use of closed network connection"}
⠇ Dumping tables ...2023-09-14T10:41:16+02:00	error	error dumping table	{"error": "write tcp 127.0.0.1:53461->127.0.0.1:53418: use of closed network connection"}
Dumping is finished! (elapsed time: 4.884509083s)
```

Anyone experiencing the same?

**Comment by orware** [comment]
Hello @DaliborPan,

You may try passing in the `--threads` optional parameter here and dropping the number down in order to help avoid the `EOF` error here if this is still occurring for you.

By default that parameter is defaulting to [16 threads](https://github.com/planetscale/cli/blob/main/internal/cmd/database/dump.go#L56) when using the `pscale database dump` command so you can try dropping it to 4 or 8 and see how the behavior is in those cases.

Are you on one of our Hobby plans by any chance or have added a significant amount of data in to your database? Those may be factors as to why it may have been working previously on September 5th but is now running into an issue compared to that earlier attempt.

**Reply by DaliborPan** [reply]
Hello @orware,

thank you for your reply! Lowering number of threads doesn't work for me. 

I'm on hobby plan, latest version of planetscale cli. For testing purpose, I've created fresh new database, added only one table with a single column, and it still doesn't work..

I also found an [issue](https://github.com/planetscale/cli/issues/730) created few days ago, so I thought that more people might be experiencing the same.

![image](https://github.com/planetscale/discussion/assets/72815195/1463ee43-2caa-4af4-a6e8-badd395aa057)

![Screenshot 2023-09-14 at 18 24 50](https://github.com/planetscale/discussion/assets/72815195/d1e5977a-3e56-4e98-96ae-f0a4699058f3)

**Reply by orware** [reply, marked as answer]
Hello @DaliborPan and @pjfancher,

We'll likely follow-up in the official issue you linked to above but we are currently investigating this further on our end so that's the quick update I can share!

I was able to recreate the issue in a new Hobby plan database so that helps us to get an idea for where the issue might be so our team will be reviewing those details to identify any fixes that may be required on our end.

**Comment by pjfancher** [comment]
I'm having the exact same problem. I'm using `pscale database dump` and the command was working fine yesterday. No big changes to the database. The database is 146MB. The --threads parameter does not work.

<img width="871" alt="Screen Shot 2023-09-14 at 1 09 22 PM" src="https://github.com/planetscale/discussion/assets/502590/f0b2b825-766d-4ddd-9b43-20b4407b233b">

**Comment by mscoutermarsh** [comment]
Update: this has been fixed!

**Reply by Qavi-Nizamani** [reply]
It is not fixed yet

`⠸ Dumping tables ...2023-10-12T13:44:26+05:00	error	error dumping table	{"error": "EOF"}
Dumping is finished! (elapsed time: 15m41.672388558s)
`

**Reply by orware** [reply]
Hello @Qavi-Nizamani, 

Based on it taking close to 16 minutes before the `EOF` error was triggered, the issue in your case is likely a little different and may be based on your current data size and a few other factors. 

You can try dropping the number of threads down from the default of 16 using the optional `--threads` parameter as an initial test but if you continue to have difficulty please send in a quick ticket over at https://support.planetscale.com/ and we can take a closer look at the details for your database and figure out a solution.
